ZenPacks/zenoss/WSMAN/modeler/WSMANPlugin.py

```diff
@@ -93,14 +95,25 @@ def collect(self, device, log):
         conn_info = self.conn_info(device)
         client = self.client(conn_info)
 
-        try:
-            results = yield client.do_enumerate(
-                map(self.create_enum_info,
-                    self.wsman_queries,
-                    self.wsman_queries.values()))
-
-        except txwsman.util.RequestError as e:
-            log.error(e[0])
-            raise
+        enum_infos = map(
+            create_enum_info,
+            self.wsman_queries,
+            self.wsman_queries.values())
+
+        results = {}
+
+        for enum_info in enum_infos:
+            try:
+                results[enum_info.className] = yield client.enumerate(
+                    enum_info.className,
+                    wql=enum_info.wql,
+                    namespace=enum_info.namespace,
+                    maxelements=device.zWSMANMaxElements)
+            except txwsman.util.RequestError as e:
+                if 'unauthorized' in e.message:
+                    raise
+                else:
+                    log.error(e.message)
```

Comment:
It would be better to prefix this log with the device.id. I realize this is just refactoring the existing log, but it's a good time to improve it.